src/DebuggerData.cpp

```diff
@@ -284,33 +284,44 @@ int Breakpoints::breakpointCount()
 	return breakpoints.size();
 }
 
-bool Breakpoints::isBreakpoint(quint16 addr, QString *id)
+bool Breakpoints::isBreakpoint(quint16 addr, QString* id, bool checkSlot)
 {
-	for (const auto& bp : breakpoints) {
-		if (bp.type == BREAKPOINT && bp.address == addr) {
-			if (inCurrentSlot(bp)) {
+	int tmp = currentIndex;
```

Comment:
This is more like critique on the code before this commit. But is does also make your PR more complicated. So consider changing it.

The methods `findBreakPoint()` and `findNextBreakPoint()` implement a 'stateful' interface. That is: they store `currentIndex` as a member variable. I think it's easy to transform this into a 'stateless' interface: remove the `currentIndex` member variable, return it from `findBreakPoint()` (already done) and require it as an input parameter in `findNextBreakpoint()`. In other words: move this state from the `BreakPoints` class to (a local variable in) the caller(s) of these methods.

This transformation allows 'concurrent' uses of this interface. And then you don't have to store `currentIndex` into `tmp`, and restore it on two different locations (error prone) in this function.

Reply:
properly fixed somewhere else